adapters/solana/account_proof_geyser/src/lib.rs

```diff
@@ -0,0 +1,471 @@
+pub mod config;
+pub mod types;
+pub mod utils;
+
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::str::FromStr;
+use std::sync::atomic::{AtomicU8, Ordering};
+use std::thread;
+
+use borsh::BorshSerialize;
+use crossbeam_channel::{unbounded, Sender};
+use log::error;
+use solana_geyser_plugin_interface::geyser_plugin_interface::{
+    GeyserPlugin, GeyserPluginError, ReplicaAccountInfoVersions, ReplicaBlockInfoVersions,
+    ReplicaEntryInfoVersions, ReplicaTransactionInfoVersions, Result as PluginResult, SlotStatus,
+};
+use solana_sdk::clock::Slot;
+use solana_sdk::hash::{hashv, Hash};
+use solana_sdk::pubkey::Pubkey;
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpListener;
+use tokio::sync::broadcast;
+
+use crate::config::Config;
+use crate::types::{
+    AccountHashAccumulator, AccountInfo, BankHashProof, BlockInfo, GeyserMessage, SlotInfo,
+    TransactionInfo, TransactionSigAccumulator, Update,
+};
+use crate::utils::{
+    assemble_account_delta_proof, calculate_root_and_proofs, get_keys_for_non_inclusion_inner,
+    get_proof_pubkeys_required, hash_solana_account,
+};
+
+fn handle_confirmed_slot(
```

Comment:
This function is huge and pretty complex. Can we add a doc comment explaining what this method does (maybe as a list of steps), and then mark each step with a comment in the body

Reply:
added the necessary documentation!